src/commands/mod.rs

```diff
@@ -436,6 +445,30 @@ impl DaemonControl {
         let mut db_conn = self.db.connection();
         db_conn.delete_spend(txid);
     }
+
+    /// Finalize and broadcast this stored Spend transaction.
+    pub fn broadcast_spend(&self, txid: &bitcoin::Txid) -> Result<(), CommandError> {
+        let mut db_conn = self.db.connection();
+
+        log::debug!("A");
+
+        // First, try to finalize the spending transaction with the elements contained
+        // in the PSBT.
+        let mut spend_psbt = db_conn
+            .spend_tx(txid)
+            .ok_or(CommandError::UnknownSpend(*txid))?;
+        log::debug!("B");
+        miniscript::psbt::finalize(&mut spend_psbt, &self.secp)
+            .map_err(|e| CommandError::SpendFinalization(e.to_string()))?;
+
+        // Then, broadcast it (or try to, we never know if we are not going to hit an
+        // error at broadcast time).
+        let final_tx = spend_psbt.extract_tx();
+        match self.bitcoin.broadcast_tx(&final_tx) {
+            Ok(()) => Ok(()),
+            Err(BitcoinError::Broadcast(e)) => Err(CommandError::TxBroadcast(e)),
```

Comment:
you can use map_err or do a custom `From`  for CommandError

Reply:
That's intentional. The advantage of using `match` here is that when/if we introduce a new `BitcoinError` variant in the future that should not be reachable from there, we can treat is as such.